webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -4160,7 +4160,38 @@ private String findSites(Map propertyCriteria) {
         return Xml.writeDocumentToString(dom);
     }
 
-
+    /**
+     * Get all site IDs for which the criteria fully or partially matches the title, the description or the skin.
+     *
+     * @param sessionid     valid session
+     * @param criteria      string to search for
+     * @return The site IDs
+     */
+    @WebMethod
+    @Path("/findSitesByTitle")
+    @Produces("text/plain")
+    @GET
+    public String findSitesByTitle(
+            @WebParam(name = "sessionid", partName = "sessionid") @QueryParam("sessionid") String sessionid,
+            @WebParam(name = "criteria", partName = "criteria") @QueryParam("criteria") String criteria) {
+        Session s = establishSession(sessionid);
+        String siteIDs = "";
+        try {
+            List<String> siteIdsList = siteService.getSiteIds(SelectionType.ANY, null, criteria,
+                    null, SortType.NONE, null);
+            StringBuilder sb = new StringBuilder();
```

Comment:
StringBuilder sb never used outside the next block

Reply:
True. I put it inside the block.